### PR TITLE
Add support for calling delivery via gRPC

### PR DESCRIPTION
**Pull request by prm-james-hill**

### CODE REVIEW

File: src/main/java/ai/promoted/delivery/client/ApiDelivery.java [hunk 1]
```diff
@@ -75,9 +102,19 @@ public ApiDelivery(String endpoint, String apiKey, long timeoutMillis, boolean w
 
     this.httpClient = HttpClient.newBuilder().version(HttpClient.Version.HTTP_2).build();
     
+    // The gRPC path doesn't support health checks or warm-up yet, so we re-use the HTTP path.
     if (warmup) {
       runWarmup();
     }
+
+    if (useGrpc) {
```
Comment (prm-dan) [inline src/main/java/ai/promoted/delivery/client/ApiDelivery.java:110]: If clients object to grpc, they probably don't want the dependency in their binary.

Long-term, I'd either do:
1. moving everyone over to grpc.
2. having different builds for grpc vs json.

If you want it temporarily, that's fine.

File: src/test/java/ai/promoted/delivery/client/IntegrationTest.java [hunk 2]
```diff
@@ -39,6 +39,29 @@ public void testCallDeliver() throws Exception {
...
+    myProps.put("search", searchProps);
+    
+    Request.Builder reqBuilder = Request.newBuilder().setUserInfo(UserInfo.newBuilder().setAnonUserId("12355")).setPlatformId(0)
+        .setPaging(Paging.newBuilder().setOffset(0).setSize(100))
+        .addInsertion(InsertionFactory.createInsertionWithProperties("28835", myProps))
+        .addInsertion(Insertion.newBuilder().setContentId("49550"));
+    add100Insertions(reqBuilder);
+    
+    DeliveryResponse resp = client.deliver(new DeliveryRequest(reqBuilder, null, false, 0));
+    System.out.println(resp);
```
Comment (prm-dan) [inline src/test/java/ai/promoted/delivery/client/IntegrationTest.java:61]: Remove the println from the test.

**Review by prm-dan — APPROVED** [review]

File: src/test/java/ai/promoted/delivery/client/IntegrationTest.java (hunk 2 above)
Comment (prm-james-hill) [reply src/test/java/ai/promoted/delivery/client/IntegrationTest.java]: Done

File: src/main/java/ai/promoted/delivery/client/ApiDelivery.java (hunk 1 above)
Comment (prm-james-hill) [reply src/main/java/ai/promoted/delivery/client/ApiDelivery.java:110]: Yeah, a more proper thing here (for the ecosystems I'm familiar with) would be conditional builds but it either doesn't seem like a thing in Java or it gets messy. I can see users needing additional functionality added to the SDK but not being able to move to gRPC so I don't want to require a move at this point.

**Pull request merged**